opta/core/kubernetes.py

```diff
@@ -387,12 +394,102 @@ def _aws_set_kube_config(layer: "Layer") -> None:
     return
 
 
-def get_cluster_name(layer: "Layer") -> Optional[str]:
-    outputs = get_terraform_outputs(layer)
-    cluster_name = outputs.get("parent.k8s_cluster_name") or outputs.get(
-        "k8s_cluster_name"
-    )
-    return cluster_name
+def get_cluster_name(layer: "Layer") -> str:
+    return f"opta-{layer.root().name}"
+
+
+def does_cluster_exist(layer: "Layer") -> bool:
+    if layer.is_stateless_mode() is True:
+        if logger.isEnabledFor(DEBUG):
+            logger.debug(
+                "does_cluster_exist called in stateless mode, verify implementation. See stack trace below:"
+            )
+            traceback.print_stack()
+
+    if layer.cloud == "aws":
+        return _aws_does_cluster_exist(layer)
+    elif layer.cloud == "google":
+        return _gcp_does_cluster_exist(layer)
+    elif layer.cloud == "azurerm":
+        return _azure_does_cluster_exist(layer)
+    elif layer.cloud == "local":
+        return _local_does_cluster_exist(layer)
+    else:
+        raise Exception(
+            f"Unknown cloud {layer.cloud}. Can not handle determining if cluster exists"
+        )
```

Comment:
it's interesting, I didn't think we would change this part- we could still rely on terraform output to know if the cluster exist.
the issue we had was that it was bundled into `get_cluster_name` and would return None if terraform output didn't exist.

Did you find a corner case where we need to know if the cluster exist but where we won't have the terraform output?

Reply:
That's actually why we had the issue in the first place-- we cached terraform state from before the k8s cluster creation, and so when we checked the output we thought that the cluster was not created. I thought we also agreed to make this does_cluster_exist function and that it must not rely on the terraform output?

Reply:
yes, I remember now - we have this issue where the terraform output is saved in memory, so calling the cloud api is good 👍 thanks for the refresher